scripts/newinstall.sh

```diff
@@ -47,44 +49,143 @@ else
 	CURL=${CURL:-curl}
 fi
 
+print_error() {
+	>&2 echo -e "$@"
+}
+
+fail() {
+	code=${2:1}
+	print_error "$1"
+	# shellcheck disable=SC2086
+	exit $code
+}
+
+miniconda::install() {
+	local python_version=$1
+	local version=$2
+	local prefix=$3
+	local miniconda_base_url=${4:-https://repo.continuum.io/miniconda}
+
+	[[ -z $python_version ]] && fail "python_version param is required"
+	[[ -z $version ]] && fail "version param is required"
+	[[ -z $prefix ]] && fail "prefix param is required"
+
+	case $(uname -s) in
+		Linux*)
+			ana_platform="Linux-x86_64"
+			;;
+		Darwin*)
+			ana_platform="MacOSX-x86_64"
+			;;
+		*)
+			fail "Cannot install miniconda: unsupported platform $(uname -s)"
+			;;
+	esac
+
+	miniconda_file_name="Miniconda${python_version}-${version}-${ana_platform}.sh"
+	echo "::: Deploying ${miniconda_file_name}"
+	$cmd $CURL -# -L -O "${miniconda_base_url}/${miniconda_file_name}"
+
+	$cmd bash "$miniconda_file_name" -b -p "$prefix"
+
+	if [[ $(uname -s) = Darwin* ]]; then
+		# run install_name_tool on all of the libpythonX.X.dylib dynamic
+		# libraries in miniconda
+		for entry in $prefix/lib/libpython*.dylib; do
```

Comment:
I don't know -- I was copying what the miniconda2/3 products were doing.  I can remove it to see if anything breaks.